src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -554,96 +571,100 @@ impl<S: S3BackEnd + Send + Sync + 'static> S3MetaData<S> {
-                            and ino={} from parent directory ino={}",
+                 and ino={} from parent directory ino={}",
                 node_name, ino, parent_ino,
             ))?;
             debug!(
                 "may_deferred_delete_node_helper() successfully remove entry name={:?} \
-                    ino={} from parent directory ino={}",
+                 ino={} from parent directory ino={}",
                 node_name, ino, parent_ino
             );
             debug_assert_eq!(node_name, deleted_entry.entry_name());
             debug_assert_eq!(deleted_entry.ino(), ino);
         }
 
         if deferred_deletion {
-            // deferred deletion
+            // Deferred deletion
             // TODO: support thread-safe
-            let (node_name, node_open_count, node_lookup_count) = {
-                let cache = self.cache.read().await;
-                let node = cache.get(&ino).unwrap_or_else(|| {
-                    panic!(
-                        "impossible case, may_deferred_delete_node_helper() \
-                        should already find the i-node of ino={} to remove",
-                        ino,
-                    );
-                });
-                (
-                    node.get_name().to_owned(),
-                    node.get_open_count(),
-                    node.get_lookup_count(),
-                )
-            };
-            {
-                let mut trash = self.trash.write().await;
-                let insert_result = trash.insert(ino); // check thread-safe in case of deferred deletion race
-                debug_assert!(
-                    insert_result,
-                    "failed to insert i-node of ino={} into trash for deferred deletion",
+            let cache = self.cache.read().await;
+            let inode = cache.get(&ino).unwrap_or_else(|| {
+                panic!(
+                    "impossible case, may_deferred_delete_node_helper() \
+                     i-node of ino={} is not in cache",
                     ino,
                 );
-            }
+            });
             debug!(
                 "may_deferred_delete_node_helper() defered removed \
                     the i-node name={:?} of ino={} under parent ino={}, \
                     open count={}, lookup count={}",
-                node_name, ino, parent_ino, node_open_count, node_lookup_count,
+                inode.get_name(),
+                ino,
+                parent_ino,
+                inode.get_open_count(),
+                inode.get_lookup_count(),
             );
+            inode.mark_deferred_deletion();
+            // Notify kernel to drop cache
+            if from_remote && inode.get_lookup_count() > 0 {
+                let fuse_fd = *self.fuse_fd.lock().await;
+                // fuse_fd must be set
+                assert!(fuse_fd > 0_i32);
+                #[cfg(feature = "abi-7-18")]
```

Comment:
We can wrap two lines into a block and write cfg once.

Reply:
Yes.